frontend/src/views/CRAllotment.vue

```diff
@@ -0,0 +1,194 @@
+              </div>
+            </template>
+          </template>
+          <p v-else>No spare files found for selected language in {{filter.list}} list.</p>
+        </template>
+        <p v-else>Loading files…</p>
+      </template>
+      <p v-else>Choose list and language to select files.</p>
+      <!-- Comment -->
+      <v-textarea v-model="allotment.comment" box label="Comment" rows="3"></v-textarea>
+      <!-- Buttons -->
+      <v-btn @click="allot" :loading="submissionStatus === 'inProgress'">Allot</v-btn>
+    </v-form>
+    <v-alert
+      v-else
+      :value="submissionStatus === 'complete'"
+      type="success"
+      transition="scale-transition"
+    >
+      <h4 class="alert-heading">Allotted successfully</h4>
+      <p class="mb-0">
+        <v-btn @click="reset">Make another allotment</v-btn>
+      </p>
+    </v-alert>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Watch } from "vue-property-decorator";
+import firebase from "firebase/app";
+import "firebase/functions";
+import _ from "lodash";
+
+import { initialAllotment } from "@/utility";
+
+@Component({
+  name: "CRAllotment"
+})
+export default class CRAllotment extends Vue {
+  assignees: any = null;
+  languages: string[] = ["English", "Hindi", "Bengali", "None"];
+  lists = null;
+  files: any = null;
+  filter = CRAllotment.initialFilter();
+  allotment: any = initialAllotment();
+  submissionStatus: string | null = null;
+
+  static initialFilter() {
+    return {
+      languages: [] as string[],
+      list: null
+    };
+  }
+
+  async mounted() {
+    this.getAssignees();
+    this.getLists();
+  }
+
+  async getAssignees() {
+    const assignees = await firebase
+      .functions()
+      .httpsCallable("User-getAssignees")({ phase: "CR" });
+
+    this.assignees = assignees.data;
+    if (this.$route.query.emailAddress) {
+      this.allotment.assignee = this.assignees.find(
+        (assignee: any) =>
+          assignee.emailAddress === this.$route.query.emailAddress
+      );
+    }
+  }
+
+  async getLists() {
+    this.filter.languages = this.languages;
+    const lists = await firebase.functions().httpsCallable("CR-getLists")();
+
+    this.lists = lists.data;
+  }
+
+  get assigneeHint() {
+    const languages = _.get(this.allotment, "assignee.languages", []);
+    const hint = languages.join(", ");
+    return hint ? `Languages: ${hint}` : "";
+  }
+
+  @Watch("allotment.assignee")
+  handleAllotmentAssignee(newValue: any) {
+    if (newValue === null) return;
+
+    this.filter.languages = this.languages;
+  }
+
+  debouncedFilter = _.debounce(async () => {
+    this.files = null;
+    this.allotment.files = [];
+    if (this.filter.list === null) return;
+
+    const spareFiles = await firebase
+      .functions()
+      .httpsCallable("CR-getSpareFiles")(this.filter);
+
+    this.files = spareFiles.data;
+  }, 1000);
+
+  @Watch("filter", { deep: true })
+  handleFilter() {
+    this.debouncedFilter();
```

Comment:
What are the benefits of extracting this as a separate function? The function is used only once, and I think it complicates understanding the code. Please consider moving the code inside `handleFilter`.

Reply:
Debounce returns a function to call.